monai/transforms/compose.py

```diff
@@ -136,3 +139,17 @@ def __call__(self, input_):
         for _transform in self.transforms:
             input_ = apply_transform(_transform, input_)
         return input_
+
+    def inverse(self, data, keys: Optional[Tuple[Hashable, ...]] = None):
+        if not isinstance(data, Mapping):
+            raise RuntimeError("Inverse method only available for dictionary transforms")
+        d = deepcopy(dict(data))
+        if keys:
+            keys = ensure_tuple(keys)
```

Comment:
Which assumptions? I checked for dictionary data because only dictionary transforms have been made invertible. Would you prefer something like this:
```python
if sum(isinstance(t, InvertibleTransforms) for t in self.flatten().transforms) == 0:
    warning.warn("inverse has been called but no invertible transforms have been supplied")
```

Reply:
sorry I didn't make it clear, I mean all the input validadtions, the method could be:
```py
def inverse(self, data, keys: Optional[Tuple[Hashable, ...]] = None):
       d = deepcopy(data)
       for t in reversed(self.transforms):
            # check if transform is one of the invertible ones
            if isinstance(t, InvertibleTransform):
                d = t.inverse(d, keys)
        return d
```

(btw are `flatten()` and `__len__` needed here? I could find where they are used...)

Reply:
Ok, but if a user has array transforms, they will call the `inverse` method but their data will remain unchanged and they won't receive any explanation as to why.

Reply:
I see, I was thinking about making`Compose.__call__` and `Compose.inverse` mostly symmetric... that would be nice :)

Reply:
Alright, I'll remove the checks and add the suggested warning. That way, apart from the warning they're symmetric. Is that OK?

Reply:
sure, thanks! still, `Compose.__call__` works fine with both array and dict, but why does `Compose.inverse` have the concept of `data` and `keys`? `InvertibleTransform` perhaps should inherit `MapTransform` since it's handling dictionaries? what do you think?

Reply:
I see your objection. I wanted to potentially be able to call the inverse on a subset of the keys. E.g., the forward pass contains image and label, then you infer a segmentation and only want to inverse the segmentation using the label's transformations. 

I think it's definitely a good idea to change `InvertibleTransform` to `InvertibleMapTransform` and have it inherit from `MapTransform`. However, I don't think that will solve this problem...

Reply:
Ok, I've removed the concept of keys from the inverse method, including `Compose.inverse`. I've replaced it with a withable [here](https://github.com/Project-MONAI/MONAI/pull/1663/files#diff-96cd15307195a8d855e3fb33c35abaa18afaecc182874751af4eb3144726d158R712):
```python
data = {"image": np.arange(16, dtype=float).reshape(1, 4, 4)}
t = SpatialPadd(["image", "label"], 10, allow_missing_keys=False)
_ = t(data)  # would raise exception
with AllowMissingKeysMode(t):
    _ = t(data)  # OK!
```
This means I can do the inverse with a subset of keys. Might also be useful in other circumstances!

What do you think? I'll put it into its own PR if you approve.